### PR TITLE
Full ordinal support in GetArrayItem

### LINKED ISSUE #8225: [BUG] GpuGetArrayItem only supports ints as the ordinal.
**Describe the bug**
The GpuGetArrayItem only supports ints as the ordinal type, but if you look at Spark the ordinal type is actually `IntegralType` which is byte, short, int, and long.

https://github.com/apache/spark/blob/699b9a5ff06c9640ff25ddd4dd47084781b556b6/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala#L246

We should support all of the types. The smaller one can just be upcast, but the long might need a bit of help to avoid overflow problems when casting so we always get the correct answer.

This should also add tests to verify that we are doing the right thing.

### PR DESCRIPTION
**Pull request by revans2**
This fixes #8225

### CODE REVIEW

File: integration_tests/src/main/python/array_test.py [hunk 1]
```diff
@@ -89,19 +88,32 @@
         DoubleGen(special_cases=[], no_nans=True),
         string_gen, boolean_gen, date_gen, timestamp_gen]
 
-# Merged "test_nested_array_item" with this one since arrays as literals is supported
+
+byte_array_index_gen = ByteGen(min_val=-25, max_val=25, special_cases=[None])
+short_array_index_gen = ShortGen(min_val=-25, max_val=25, special_cases=[None])
+int_array_index_gen = IntegerGen(min_val=-25, max_val=25, special_cases=[None])
+long_array_index_gen = LongGen(min_val=-25, max_val=25, special_cases=[None])
```
Comment (jlowe) [inline integration_tests/src/main/python/array_test.py:95]: Is it worth adding a special case where we use a large long that will truncate to a valid index when truncated to an int?  This is to check for any regressions or change in behavior regarding handling of long indices.

**Review by jlowe — DISMISSED** [review]
Looks good to me, just a small suggestion for testing a corner case with long indices.

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@jlowe please take another look

**Comment by revans2** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**